src/main/java/seedu/address/logic/commands/AddRestaurantNoteCommand.java

```diff
@@ -63,61 +65,46 @@ public AddRestaurantNoteCommand(Index index, Note recommendedFood, Note goodFood
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
         List<Restaurant> lastShownList = model.getFilteredRestaurantList();
 
         if (index.getZeroBased() >= lastShownList.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_RESTAURANT_DISPLAYED_INDEX);
         }
 
         Restaurant restaurantToEdit = lastShownList.get(index.getZeroBased());
-
-        ArrayList<Note> updatedRecFood = new ArrayList<>();
-        for (Note i : restaurantToEdit.getRecommendedFood()) {
-            updatedRecFood.add(i);
-        }
-
-        ArrayList<Note> updatedGoodFood = new ArrayList<>();
-        for (Note i : restaurantToEdit.getGoodFood()) {
-            updatedGoodFood.add(i);
-        }
-
-        ArrayList<Note> updatedBadFood = new ArrayList<>();
-        for (Note i : restaurantToEdit.getBadFood()) {
-            updatedBadFood.add(i);
+        if (recommendedFood.size() == 0 || goodFood.size() == 0 || badFood.size() == 0) {
+            throw new CommandException(MESSAGE_EMPTY);
```

Comment:
Shouldn't this be if (recommendedFood.size() == 0 && goodFood.size() == 0 && badFood.size() == 0 instead? O:

Reply:
The user might have added notes, for example, to recommend & good food but missed out bad food. So if any of the notes to be added are empty, it will throw an error message.